backend/routes/league.go

```diff
@@ -1,11 +1,94 @@
 package routes
 
-// s := r.PathPrefix("/league").Subrouter()
-// // "/products/"
-// s.HandleFunc("/postleague", AddLeague)
-// // "/products/{key}/"
-// s.HandleFunc("/getLeagueById/{ID}", GetLeagueById)
-// // "/products/{key}/details"
-// s.HandleFunc("/deleteLeagueById/{ID}", DeleteLeague)
-// // "/league/deleteLeagueById/{ID}"
-// s.HandleFunc("/updateLeagueById/{ID}", UpdateLeague)
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/grommers00/civitas/backend/internal"
+	"github.com/grommers00/civitas/backend/models"
+
+	"github.com/gorilla/mux"
+)
+
+// ConnectLeaguesSubrouter adds the different restFUL apis to the main router
+func ConnectLeaguesSubrouter(r *mux.Router) *mux.Router {
+	// "/leagues/"
+	f := r.PathPrefix("/leagues").Subrouter()
+
+	// "/leagues/"
+	f.HandleFunc("/", AddLeagues).Methods("POST")
+
+	// "/leagues/"
+	f.HandleFunc("/", GetAllLeagues).Methods("GET")
+
+	// "/leagues/{ID}"
+	f.HandleFunc("/{ID}", GetLeaguesByID).Methods("GET")
+
+	// "/leagues/{ID}"
+	f.HandleFunc("/{ID}", DeleteLeaguesByID).Methods("DELETE")
+
+	// "/leagues/{ID}"
+	f.HandleFunc("/{ID}", UpdateLeaguesByID).Methods("PATCH")
+
+	return f
+}
+
+// AddLeagues will add a new article the list of articles
+func AddLeagues(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte("leagues Added!\n"))
```

Comment:
In `devel.go` under `internal` package, you'll see a `NotImplementedHandler` function, let's leverage that so that way all the writer logic is consolidated to a single function.

Reply:
fixed